manifests/server/resources/coredns.pp

```diff
@@ -19,6 +20,7 @@
   Optional[Array] $image_pull_secrets    = $k8s::server::resources::image_pull_secrets,
   Hash[String,Data] $deployment_config   = $k8s::server::resources::coredns_deployment_config,
   Array[String[1]] $hosts                = [],
+  String[1] $template_path               = 'k8s/server/resources/coredns_configmap.epp',
```

Comment:
I feel like this should probably be an `Optional[String[1]] $corefile_content = undef` parameter instead (with a fallback to the built-in render), to remove the internal template complexity from providing a replacement configuration.